### PR TITLE
Creating a Settings for Skate so users can choose update file to display

### PR DESCRIPTION
**Pull request by kateliu20**
Created the UI component for Skate Plugin, to make the Skate Settings visible and allow users to change the file path to display on the "What's New" Panel. It also has a checkbox that when clicked, the "What's New" panel is visible, but unclicked it is not on the toolbar side panel.

It also defaults to "CHANGELOG.md" unless user picks otherwise, and users can only select .md files

Screenshot 2023-06-26 at 3 43 24 PM

### CODE REVIEW

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (ZacSweers) [inline skate-plugin/src/main/resources/messages/skateBundle.properties:2]: Let's fix these names, as they're not baseline files

**Review by ZacSweers — COMMENTED** [review]
Let's clean up the `baseline` terminology usage from the detekt plugin

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt
Comment (arpita184) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt:27]: Let's add some description about each class

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt
Comment (arpita184) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt:35]: Let's rename this once you fix the baseline files reference

**Review by arpita184 — CHANGES_REQUESTED** [review]

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt:27]: Is a description like "manages user-specific settings for the skate plugin" sufficient?

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooserDescriptorUtil.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooserDescriptorUtil.kt:30]: Let's remove the things we're not using. Another reason to avoid copy-pasting unnecessarily :)

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FilesListPanel.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FilesListPanel.kt:32]: Let's remove this, you can suppress it in your IDE settings instead

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FilesListPanel.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FilesListPanel.kt:69]:
```suggestion
      if (file != null && file !in listModel.items) {
```

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (ZacSweers) [inline skate-plugin/src/main/resources/messages/skateBundle.properties:3]: Skate file or Changelog file?

**Review by ZacSweers — COMMENTED** [review]
For copied classes, you need to copy the original copyright header and let's link where it's from for posterity

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt
Comment (arpita184) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt:27]: As an example:

https://softwareengineering.stackexchange.com/questions/187882/how-can-i-become-better-on-explaining-the-code-to-other-developers

https://towardsdatascience.com/5-tips-to-write-better-comment-in-your-code-1d4681071967

As you are adding more code, I want you to think about readability of the code. Is it easier for someone else to read your code and makes sense of your action and intention? How your documentation can help etc.

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooserDescriptorUtil.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooserDescriptorUtil.kt:30]: Maybe I will just create a new kotlin file instead of the copied FileChooserDescriptorUtil.kt

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (kateliu20) [reply skate-plugin/src/main/resources/messages/skateBundle.properties:3]: Changelog is good, thanks

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt:21]: Let's put a doc on this. Also the name is a little confusing, we don't usually put suffixes like this. Let's just call it `FileChoosing`?

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooserDescriptorUtil.kt
Comment (ZacSweers) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooserDescriptorUtil.kt:30]: Did you mean to leave this after your update?

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt:35]: It looks like this is missing the checkbox to enable/disable whatsnew. Let's

1. add that checkbox UI
2. Make the file chooser dependent on that being enabled

Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt:35]: Okay, sounds good ! Thank you 🙏

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt:21]: When you mean doc, do you mean documentation within the skate plugin code?

Comment (ZacSweers) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt:21]: yep! Let's put a doc on this object

```suggestion
/**
 * A meaningful doc explaining what this code is for. Doesn't have to be super detailed, just enough for someone reading this to know what it's for at a glance.
 */
object FileChooseDescriptorSkate {
```

Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt]: Added a doc 👍

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt]: 887cc88

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt:28]: dd48f90

**Comment by ZacSweers** [comment]
Re this screenshot – it looks like the changelog file is actually not selectable?

image

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChooseDescriptorSkate.kt]: dd48f90

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FilesListPanel.kt
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FilesListPanel.kt]: 7e8b41f

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (kateliu20) [reply skate-plugin/src/main/resources/messages/skateBundle.properties]: Here is the commit where I changed the name: 22729e86ce009f741e96202899dd81dff1f077e6

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt:45]: nit: this is done now in your UI and we can remove the comment

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChoosing.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/FileChoosing.kt:22]: nit: formatting's off here

```suggestion
 * Provides utility function for creating file chooser descriptors to select specific types of
```

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/SkateConfigUI.kt:49]: Doesn't have to be in this PR, but assuming we'll add other settings here in the future so let's group these into a more meaningfully named setup

```kotlin
fun Panel.whatsNewSection()
fun Panel.whatsNewCheckboxRow()
fun Panel.whatsNewPathChooserRow()
```

Something like that

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (ZacSweers) [inline skate-plugin/src/main/resources/messages/skateBundle.properties:3]: In a followup, let's see if these support templating so we can reuse this rather than hardcode the "Changelog" name

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (ZacSweers) [inline skate-plugin/src/main/resources/messages/skateBundle.properties:5]: nit: let's not quote these. Also suggesting a tweak to the description

```suggestion
skate.configuration.enableWhatsNew.title=Enable What's New Panel
skate.configuration.enableWhatsNew.description=If enabled, shows the What's New Panel if there are new changes detected in a specified changelog file.
```

File: skate-plugin/src/main/resources/messages/skateBundle.properties
Comment (ZacSweers) [inline skate-plugin/src/main/resources/messages/skateBundle.properties:2]: nit: titles don't have trailing colons

```suggestion
skate.configuration.choosePath.title=Choose file
```

**Review by ZacSweers — APPROVED** [review]
Approving with nits! This means there are a few small nits I'd like to see get fixed first, but trusting you to take care of them and merge away without needing me to do another review pass 👍

**Pull request merged**
